kerchunk/combine.py

```diff
@@ -488,26 +488,78 @@ def _reorganise(coos):
         out[k] = np.array(sorted(arr))
     return out
 
+
 def merge_vars(files, storage_options=None):
     """Merge variables across datasets with identical coordinates
 
     :param files: list(dict), list(str) or list(fsspec.OpenFile)
         List of reference dictionaries or list of paths to reference json files to be merged
     :param storage_options: dict
-        Dictionary containing kwargs to `fsspec.open_files`    
+        Dictionary containing kwargs to `fsspec.open_files`
     """
     if isinstance(files[0], collections.abc.Mapping):
         fo_list = files
         merged = fo_list[0].copy()
         for file in fo_list[1:]:
-            refs = file['refs']
-            merged['refs'].update(refs)
+            refs = file["refs"]
+            merged["refs"].update(refs)
     else:
         fo_list = fsspec.open_files(files, mode="rb", **(storage_options or {}))
         with fo_list[0] as f:
             merged = ujson.load(f)
         for file in fo_list[1:]:
             with file as f:
-                refs = ujson.load(f)['refs']
-            merged['refs'].update(refs)
+                refs = ujson.load(f)["refs"]
+            merged["refs"].update(refs)
     return merged
+
+
+def concatenate_arrays(
+    files, storage_options=None, axis=0, key_seperator=".", path=None
+):
+    """Simple concatenate of one zarr array along an axis
+
+    Assumes that each array is identical in shape/type.
+
+    If the inputs are groups, provide the path to the contained array, and all other arrays
+    will be ignored. You could concatentate the arrays separately and then recombine them
+    with ``merge_vars``.
+
+    Parameters
+    ----------
+    files: list[dict] | list[str]
+        Input reference sets, maybe generated by ``kerchunk.zarr.single_zarr``
+    storage_options: dict | None
+        To create the filesystems, such at target/remote protocol and target/remote options
+    key_seperator: str
+        "." or "/", how the zarr keys are stored
+    path: str or None
+        If the
+    """
+    out = {}
+    if path is None:
+        path = ""
+    else:
+        path = "/".join(path.rstrip(".").rstrip("/").split(".")) + "/"
+
+    for i, fn in enumerate(files):
+        fs = fsspec.filesystem("reference", fo=fn, **(storage_options or {}))
+        if i == 0:
+            zdata = ujson.load(fs.open(f"{path}.zarray"))
+            shape = zdata["shape"]
+            chunks_per_file = int(shape[axis] / zdata["chunks"][axis])
+            shape[axis] *= len(files)
+            zdata["shape"] = shape
+            out[f"{path}.zarray"] = ujson.dumps(zdata)
+            for name in [".zgroup", ".zattrs", f"{path}.zattrs"]:
+                if name in fs.references:
+                    out[name] = fs.references[name]
```

Comment:
Does this mean the result will always have the `.zattrs` of the last file?

Reply:
Actually of the first dataset (i == 0). They should all be the same.

Reply:
Ah yeah, skipped that line 🤦

> They should all be the same.

Is this something that's normally checked?

Reply:
Could do, but I think a convenience function like this can leave it to the caller.

Reply:
Yeah, I think it's fine for now, but it would be nice for it to be documented.

Maybe having a callback argument like `combine_metadata(key: str, entries: list[dict]) -> dict` could work, but also totally fine to leave simple for now.

Reply:
Yeah, I don't want to add more functionality, or else we creep towards another version of MultiZarrToZarr. The function as it is is useful, but also shows the kinds of things (along with merge_variables, just above) that can be done to directly manipulate references.